cider/tests/utils/CiderBatchChecker.cpp

```diff
@@ -198,6 +198,49 @@ bool CiderBatchChecker::checkOneScalarBatchEqual<bool>(
   return true;
 }
 
+bool CiderBatchChecker::checkOneVarcharBatchEqual(const VarcharBatch* expected_batch,
+                                                  const VarcharBatch* actual_batch) {
+  if (!expected_batch || !actual_batch) {
+    std::cout << "One or more ScalarBatches are null_ptr in checkOneScalarBatchEqual. "
+              << "This can be caused by casting a ScalarBatch to a wrong type."
+              << std::endl;
+    return false;
+  }
+
+  // compare nulls
+  bool null_buffer_eq = checkValidityBitmapEqual(expected_batch, actual_batch);
+  if (!null_buffer_eq) {
+    std::cout << "Null buffer memcmp failed." << std::endl;
+    return false;
+  }
+
+  int row_num = actual_batch->getLength();
+
+  // compare offset
+  auto expected_offsets = expected_batch->getRawOffset();
```

Comment:
just curious, if we have such two vectors:
vec1_data ["a", "a","a","a", "a"]
vec1_offset[0,1,2,3,4,5]
vec1_null[T,T,F,T,T]

vec2_data ["a", "a","","a", "a"]
vec2_offset[0,1,2,2,3,4]
vec2_null[T,T,F,T,T]

should vec1 be equal to vec2?

Reply:
I think they should be considered equal. We don't need to care about differences in null values.

Also added a UT to verify this.